packages/xstate-react/src/useMachine.ts

```diff
@@ -1,14 +1,18 @@
 import { useCallback, useState } from 'react';
 import {
+  ActionFunction,
   EventObject,
-  StateMachine,
-  State,
   Interpreter,
   InterpreterOptions,
-  MachineOptions,
+  MaybeTypegenMachineOptions,
+  State,
   StateConfig,
+  StateMachine,
   Typestate,
-  ActionFunction
+  AreAllImplementationsAssumedToBeProvided,
```

Comment:
This introduces a type-level dependency on the newest version of XState. In theory, we declare that `@xstate.react` works with `xstate@>=4.11.0`:
https://github.com/statelyai/xstate/blob/385d29847a1462ee7cd564fdec0a8d66cbf34406/packages/xstate-react/package.json#L50
and this stops being true because of changes like this.

Issues like this have been reported to us in the past (like [here](https://github.com/statelyai/xstate/issues/2410)) - so it's not like a theoretical problem. People don't always upgrade `@xstate/react` together with `xstate`.

Should we care about issues like this? My general stance is that I try not to inflict such issues on users - it is quite problematic in practice though.

Note that this could be avoided by introducing a package like `@xstate/internal-types` that would be a regular dependency of both `xstate` and `@xstate/react`.

cc @davidkpiano

Reply:
One of my concerns is that the typegen types are bleeding into other XState packages (like `@xstate/react`, and will soon become necessary for others); is there any way to prevent this and have them only apply to `xstate` core? Even at the small expense of not providing as helpful typing for the XState helpers

Reply:
@davidkpiano What's the downside of the typegen types bleeding into other packages?

Reply:
It's IMHO crucial for this to be supported in the integration packages - the DX would be quite severely impacted by not including this in those packages.